torch/_inductor/compile_fx.py

```diff
@@ -681,6 +702,14 @@ def compile_fx(
 
     graph_id = next(_graph_counter)
 
+    orig_model = model_
+    if config.keep_output_stride:
+        # we need know the number of outputs of the original model.
+        # should do a make_fx if orig_model is not a GraphModule yet.
+        # This happens if people call inductor without dynamo.
+        if not isinstance(orig_model, torch.fx.GraphModule):
+            orig_model = make_fx(orig_model)(*example_inputs_)
+
```

Comment:
Doing an extra trace just for number of outputs seems overkill.  I think you can use the same code we do for cudagraphs to detect number of outputs:
https://github.com/pytorch/pytorch/blob/8a7d540b4c6eb1264806429a92c08778f0801178/torch/_inductor/compile_fx.py#L590-L611
(note the number of non-tangent inputs to the backwards is the number of outputs)

Alternatly, internally aot_autograd knows the number of outputs.

Reply:
oh I think I can not use `count_tangents` here since  the passed in `model_` is not a GraphModule while `count_tangents` requires a GraphModule input . 

If we call inductor from dynamo, we'll always pass a GraphModule  to `compile_fx`. But this is not true in some test cases here: https://github.com/pytorch/pytorch/blob/main/test/inductor/test_standalone_compile.py .  So the increased compilation time due to the extra `make_fx` call will only affect the 'standalone' compile.

We could let AOTAutograd expose the number of outputs for the original model. But that may increase code complexity. Should we either keep it as is or just disable layout optimization if the model passed to `compile_fx` is not a graph module (as @Chillee  mentioned above).

Reply:
> We could let AOTAutograd expose the number of outputs for the original model

fwiw, this info should also be available easily with Elias's PR: `torch._guards.TracingContext.get().fw_metadata.num_outputs`

Reply:
I think we should just assume all outputs need to be preserved if the model passed to `compile_fx` is not a graph module.

Reply:
ok. One side effect is layout optimization may not help much in that case since we may even force saved-activation to retain eager strides. But considering this is not the most common way people use inductor, I can do that to simply code.

We can always revisit this if later on we found this standalone usage becoming more popular and critical.